Examples/test-suite/csharp/director_basic_runme.cs

```diff
@@ -69,6 +71,14 @@ class MyOverriddenClass : MyClass {
       throw new Exception("null not received as expected");
     return b;
   }
+
+  public bool nonVirtual() {
```

Comment:
This is not a valid use case, the C# compiler warns:
```
./director_basic_runme.cs(75,15): warning CS0114: `director_basicNamespace.MyOverriddenClass.nonVirtual()' hides inherited member `director_basicNamespace.MyClass.nonVirtual()'. To make the current member override that implementation, add the override keyword. Otherwise add the new keyword
director_basic/MyClass.cs(96,23): (Location of the symbol related to previous warning)
```
Please adjust the testcase to be valid. Does the patch fix this invalid use case or a valid use case where the `new` and `override` keywords have been added? If it only fixes invalid C# usage, I'll be inclined to reject it as the C# compiler already shows there is a problem. Also, please add code to the director_basic_runme.cs that calls nonVirtual() which works now and which failed (got stuck in a recursive loop) before the patch.

Reply:
I think you need to change the C++ nonVirtual method into nonVirtual1 and add nonVirtual2 so that you have two different non-virtual methods to add to MyOverriddenClass, one should use `override` and the other `new`. Likewise for nonOverride.

Reply:
Hmm you probably mean that these additions would allow doing a test without `%typemap(cscode) MyClass` inject. Well that is a good idea!

Reply:
Err on the second thought it wont work because methods can not be overloaded based on these keywords. In this case in mot sure what would be achieved by renaming of these methods. Could you please elaborate?

Reply:
The idea is to test both `override` and `new` in the user's derived class.

Reply:
I dont follow. Swig already generates code that calls right method, however it takes a slow path. Patch makes it generate code that takes a fast path when method hiding is intended. I dont see how this timing can be tested without doing something unreliable/complex.

Reply:
You've removed the warning and replaced with good C# code, so am okay with this change. There are lots of ways that the director code can be called. I've used some clues about what you said and replicated the fast/slow path and added this via the calls to `call_nonVirtual` in commit  b90e3ae35d9a9c6ccb5119cb4b84b288baa75443. That's what I was looking for to make the review of exactly what you are fixing easier.